mlflow/utils/_capture_modules.py

```diff
@@ -95,6 +95,27 @@ def __exit__(self, *_, **__):
         importlib.import_module = self.original_import_module
 
 
+class _CaptureImportedModulesForHF(_CaptureImportedModules):
+    """
+    A context manager to capture imported modules by temporarily applying a patch to
+    `builtins.__import__` and `importlib.import_module`.
+    Used for 'transformers' flavor only.
+    """
+
+    def __init__(self, disable_tf=False):
+        super().__init__()
+        self.disable_tf = disable_tf
+        self.disable_torch = not self.disable_tf
+
+    def _wrap_package(self, name):
+        if (self.disable_tf and name == "tensorflow") or (self.disable_torch and name == "torch"):
+            raise ImportError
```

Comment:
```suggestion
            raise ImportError("...")
```

I think we should add a message here to only catch and suppress this error.